subsys/usb/device_next/usbd_device.h

```diff
@@ -79,9 +79,9 @@ static inline void usbd_status_suspended(struct usbd_contex *const uds_ctx,
  *
  * @param[in] node Pointer to a device context
  */
-static inline void usbd_device_lock(struct usbd_contex *const uds_ctx)
+static inline int usbd_device_lock(struct usbd_contex *const uds_ctx)
 {
-	k_mutex_lock(&uds_ctx->mutex, K_FOREVER);
+	return k_mutex_lock(&uds_ctx->mutex, K_NO_WAIT);
```

Comment:
Is this necessary? This puts quite big responsibility on checking for `-EBUSY` on the callers.